server/src/main/java/io/crate/expression/predicate/NotPredicate.java

```diff
@@ -229,10 +229,16 @@ public Query toQuery(Function input, LuceneQueryBuilder.Context context) {
                 // result set of the query
                 var refExistsQuery = IsNullPredicate.refExistsQuery(
                     nullableRef,
-                    context,
-                    countEmptyArrays(context.parentQuery(), nullableRef, context));
+                    context
+                );
                 if (refExistsQuery != null) {
                     builder.add(refExistsQuery, BooleanClause.Occur.MUST);
+                } else {
+                    // fall back
+                    return new BooleanQuery.Builder()
+                        .add(notX, Occur.MUST)
```

Comment:
afaik, we add MUST before if/else a initial state of the builder
` builder.add(notX, BooleanClause.Occur.MUST);`, so we end up adding it twice.

I think in fallback we should add only `.add(LuceneQueryBuilder.genericFunctionFilter(input, context), Occur.FILTER `

Reply:
I thought we could fall back to https://github.com/crate/crate/blob/6d1907556c74f45af55a674598b6ed3b4c04b6ec/server/src/main/java/io/crate/expression/predicate/NotPredicate.java#L220-L223
instead of continuiting to build on top of https://github.com/crate/crate/blob/6d1907556c74f45af55a674598b6ed3b4c04b6ec/server/src/main/java/io/crate/expression/predicate/NotPredicate.java#L225-L226

Could you double check?

Reply:
ah right, we have a new Builder here, sorry